### DISCUSSION TITLE
Text bubble detection

### DISCUSSION THREAD

**Discussion by mizaki** [GeoRW/ACBF-Editor#23] [General]
I'm trying to re-enable to text bubble detection and I see it only detects the bubble under the mouse cursor.

Ideally, we'd want to find all the text bubbles/frames on the page in one go but I expect that's quite hard. The ways I found to do it involve "AI" or OCR (tesseract gives you bbox for text blocks for example). These two bring large libraries with them.

I wondered if you tried to only use OpenCV to find all the bubbles/boxes?

**Comment by GeoRW** [comment]
Yeah, the problem with text bubbles is that you need to detect text there first or try to guess background color maybe. With my approach, I let user choose where the text bubble is and take background color from the underlying pixel he clicked and based on that I detect the shape. It doesn't work with joined bubbles at the moment, for those I draw ellipse now. Working code is in GTK3 repo.
With CV you can create some stuff pretty quickly, but the more you try to improve it, the complexity rises exponentially and not just the code, also demand for computer resources.
